### PR TITLE
DOCSP-34038: agg tutorials landing

### PR DESCRIPTION
**Pull request by rustagir**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-node/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-34038>
Staging - https://preview-mongodbrustagir.gatsbyjs.io/node/DOCSP-34038-tutorials-landing/aggregation-tutorials/

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/aggregation-tutorials.txt [hunk 1]
```diff
@@ -1,12 +1,104 @@
...
+   :copyable: false
+
+   const uri = "mongodb+srv://mongodb-example:27017";
+
+To run the completed file after you modify the template for a
+tutorial, run the following command in your terminal:
+
+.. code-block:: bash
+
+   node agg_tutorial.js
```
Comment (rachel-mack) [inline source/aggregation-tutorials.txt:99]: Note that running this script before making changes per an individual tutorial content will generate an error.

Comment (rachel-mack) [inline source/aggregation-tutorials.txt:99]: Note that running this script before making changes per an individual tutorial will generate a connection error.

File: source/aggregation-tutorials.txt [hunk 2]
```diff
@@ -1,12 +1,104 @@
...
+
+.. important::
+
+   In the preceding code, read the code comments to find the sections of
+   the code that you must modify for the tutorial you are following.
+
+For every tutorial, you must replace the connection string placeholder with
+your deployment's connection string. For example, if your connection string is
+``"mongodb+srv://mongodb-example:27017"``, your connection string assignment resembles
+the following:
```
Comment (rachel-mack) [inline source/aggregation-tutorials.txt:87]:
```suggestion
For every tutorial, you must replace the connection string placeholder with
your deployment's driver connection string. For example, if your connection string is
``"mongodb+srv://mongodb-example:27017"``, your connection string assignment resembles
the following:
```
You may also want to note that the users IP addresses should be added to the Access List.

File: source/aggregation-tutorials/one-to-one-join.txt [hunk 3]
```diff
@@ -20,69 +20,79 @@ One-to-One Join
...
+another collection with the same field value. The aggregation matches
+these documents on the field value and combines information from both
+into one document.
 
 .. tip::
 
-   A one-to-one join does not require the entities to have a :wikipedia:`One-to-one relationship
-   </One-to-one_(data_model)>`.
+   A one-to-one join does not require the documents to have a
+   One-to-one relationship. To learn more about this data relationship,
```
Comment (rachel-mack) [inline source/aggregation-tutorials/one-to-one-join.txt:33]: I don't think capitalization is necessary here - only in reference to the title of the linked doc.
```suggestion
   one-to-one relationship. To learn more about this data relationship,
```

File: source/aggregation-tutorials/one-to-one-join.txt [hunk 4]
```diff
@@ -20,69 +20,79 @@ One-to-One Join
...
+includes the product details associated with each order.
 
 This example uses two collections:
 
-- ``products`` which contains documents detailing a shop's available products.
-- ``orders`` which contains documents describing an individual order for a product at that shop.
+- ``orders``, which contains documents describing an order
+  for a product from a shop
+- ``products``, which contains documents that describe the available
+  products from a shop
```
Comment (rachel-mack) [inline source/aggregation-tutorials/one-to-one-join.txt:50]: Nit:
```suggestion
  products in a shop
```

File: source/aggregation-tutorials/one-to-one-join.txt [hunk 5]
```diff
@@ -20,69 +20,79 @@ One-to-One Join
...
 
 Before You Get Started
 ----------------------
 
-Before you get started with this tutorial, you will need a Node.js application
-with the {+driver-long+} installed that connects to a MongoDB deployment.
+Before you start following this tutorial, set up the aggregation
+tutorial template app in file called ``agg_tutorial.js`` by following the
+instructions in the :ref:`node-agg-tutorial-template-app` section of the
+Aggregation Tutorials landing page.
```
Comment (rachel-mack) [inline source/aggregation-tutorials/one-to-one-join.txt:63]:
```suggestion
Before you begin this tutorial, follow the instructions in the 
:ref:`node-agg-tutorial-template-app` section to set up the aggregation 
tutorial template app.
```

File: source/aggregation-tutorials/one-to-one-join.txt [hunk 6]
```diff
@@ -20,69 +20,79 @@ One-to-One Join
...
 
-To learn how to create a MongoDB deployment on MongoDB Atlas, or
-on a local MongoDB instance, see the :ref:`Connection Guide <node-connect-to-mongodb>`.
+Get a handle on the ``products`` and ``orders`` collections:
```
Comment (rachel-mack) [inline source/aggregation-tutorials/one-to-one-join.txt:65]: Maybe you're using standard phrasing I'm unfamiliar with, but I think this reads clearer:

```suggestion
Create handles for the ``products`` and ``orders`` collections:
```

File: source/includes/aggregation/one-to-one-join.js [hunk 7]
```diff
@@ -1,56 +1,19 @@
 const { MongoClient } = require("mongodb");
 
-// Replace the URI string with your connection string.
-const uri = "<Your MongoDB URI>";
+const uri = "<connection string>";
```
Comment (rachel-mack) [inline source/includes/aggregation/one-to-one-join.js:3]:
```suggestion
const uri = "<drivers connection string>";
```

**Review by rachel-mack — COMMENTED** [review]
Some suggestions for minor changes. The tutorial worked as expected following the instructions from start to finish, including set up. 👍

File: source/aggregation-tutorials.txt (hunk 1 above)
Comment (rachel-mack) [reply source/aggregation-tutorials.txt:99]: For reference:
Screenshot 2023-11-17 at 4 11 22 PM

File: source/aggregation-tutorials.txt (hunk 2 above)
Comment (rustagir) [reply source/aggregation-tutorials.txt:87]: What do you mean by access list?

Comment (rustagir) [reply source/aggregation-tutorials.txt:87]: Also, I don't think connection strings are usually driver specific, so I don't think we should describe it as a driver connection string.

File: source/aggregation-tutorials/one-to-one-join.txt (hunk 6 above)
Comment (rustagir) [reply source/aggregation-tutorials/one-to-one-join.txt:65]: changed wording

File: source/includes/aggregation/one-to-one-join.js (hunk 7 above)
Comment (rustagir) [reply source/includes/aggregation/one-to-one-join.js:3]: see my reply to your earlier comment

File: source/aggregation-tutorials.txt (hunk 2 above)
Comment (rachel-mack) [reply source/aggregation-tutorials.txt:90]: >  I don't think connection strings are usually driver specific

They're not specific to an individual driver, but from my understanding, there are different types of connection strings for different types of access. So in this case, you want the string intended for drivers. This is the screen I'm thinking of, and where I got `driver` connection string from:
Screenshot 2023-11-20 at 12 14 47 PM

>What do you mean by access list?

I mean adding the IP address you'll be connecting from, to this list. I might be wrong tho, maybe this isn't required?

Screenshot 2023-11-20 at 12 18 55 PM

Comment (rustagir) [reply source/aggregation-tutorials.txt:90]: I believe configuring the access list is optional. Also, I think the Atlas connect modal just has the "drivers" option for connecting to show you additional steps to install the driver and the connection code -- even so, the connection string is not necessarily unique to drivers. I have used my connection string to connect by using a driver, but also for connecting within Compass and by using the shell.

**Review by jordan-smith721 — APPROVED** [review]
Just a few suggestions, but otherwise LGTM!

**Pull request merged**